chain/beacon/node.go

```diff
@@ -142,6 +142,11 @@ func (h *Handler) ProcessPartialBeacon(c context.Context, p *proto.PartialBeacon
 	}
 
 	nodeName := node.Address()
+	if nodeName == h.addr {
```

Comment:
is this really correct? As I understand it, `node.Address()` will return the path (e.g. `pl-rpc1.drand.sh`), but `node.Index()` will return the index.
I think here we should be using our `node.Address()` to inspect the value `h.crypto.GetGroup()` for our index and comparing them

Reply:
Here's the flow:
https://github.com/drand/drand/blob/8ad9406fbdb54c3f7a3669663869d004b76c9978/chain/beacon/node.go#L133
is retrying the index of the partial received.
Then we retrieve the corresponding node from the group file:
https://github.com/drand/drand/blob/8ad9406fbdb54c3f7a3669663869d004b76c9978/chain/beacon/node.go#L138

We get this node's address:
https://github.com/drand/drand/blob/8ad9406fbdb54c3f7a3669663869d004b76c9978/chain/beacon/node.go#L144

and if it matches ours we don't do anything:
https://github.com/drand/drand/blob/8ad9406fbdb54c3f7a3669663869d004b76c9978/chain/beacon/node.go#L145-L146

So we are effectively comparing whether the advertised index in the partial matches our own node or not.

Reply:
Ahhh I understand - so the node that receives a partial from its own index will report `wtf?` and the other nodes will still produce `bls: signature invalid`? Should help in diagnosis at least

Reply:
Yes, exactly